src/nvim/arglist.c

```diff
@@ -346,12 +346,7 @@ static void alist_add_list(int count, char **files, int after, bool will_edit)
   int old_argcount = ARGCOUNT;
   ga_grow(&ALIST(curwin)->al_ga, count);
   if (check_arglist_locked() != FAIL) {
-    if (after < 0) {
-      after = 0;
-    }
-    if (after > ARGCOUNT) {
-      after = ARGCOUNT;
-    }
+    after = MAX(MIN(after, ARGCOUNT), 0);
```

Comment:
Would it make sense to add a `CLAMP(foo, a, b)` macro for this? (Right now it's probably not used often enough to be worth it, but I personally find that easier to grok.)

Reply:
Yeah I considered that. I went with just using MIN/MAX as they are semi-std functions, but looks like we may need ,`min()` and `max()` anyway.

Reply:
Hmm thinking about this, I think this might add a bit of ambiguity to the code as it could be interpreted as eith `CLAMP(min, foo, max)` or `CLAMP(foo, min, max)` (and any other permutation).

Maybe these cases could be written as either: `MAX(min, MIN(foo, max))` or `MIN(MAX(foo, min), max)`? So `min` always precedes `max`.

Reply:
ah, true, that would be an issue with macros -- but a function could be documented? (`clamp(foo, min, max)` is the standard signature)